File: sdk/python/tempmail_sdk/providers/j_fairuse_org.py
"""
Mailinator 姊妹域名渠道：j.fairuse.org
MX 指向 mailinator，收信复用 mailinator 的 domain=public API。
"""

from typing import List

from ..types import Email, EmailInfo
from . import mailinator

CHANNEL = "j-fairuse-org"
DOMAIN = "j.fairuse.org"


def generate_email() -> EmailInfo:
    """生成 j.fairuse.org 域名的临时邮箱"""
    local = mailinator._random_string(12)
    return EmailInfo(channel=CHANNEL, email=f"{local}@{DOMAIN}")


def get_emails(token: str, email: str = "", **kwargs) -> List[Email]:
    """收信直接委托给 mailinator（domain=public 可读所有姊妹域名邮件）"""
    return mailinator.get_emails(token, email, **kwargs)

File: sdk/npm/src/providers/j-fairuse-org.ts
import { InternalEmailInfo, Email, Channel } from "../types";
import * as mailinator from "./mailinator";
const CHANNEL: Channel = "j-fairuse-org";
const DOMAIN = "j.fairuse.org";
function randomString(l: number): string {
  const c = "abcdefghijklmnopqrstuvwxyz0123456789";
  let r = "";
  for (let i = 0; i < l; i++) r += c[Math.floor(Math.random() * c.length)];
  return r;
}
export async function generateEmail(): Promise<InternalEmailInfo> {
  return { channel: CHANNEL, email: `${randomString(12)}@${DOMAIN}` };
}
export async function getEmails(
  token: string,
  email: string,
): Promise<Email[]> {
  return mailinator.getEmails(token, email);
}
